enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -307,6 +311,26 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
 
     }
 
+    private void checkTokenEnv(JWTClaimsSet claims, String gatewayEnv) throws APISecurityException {
+        // If the claim "aud" does not exist, getAudience() returns an empty list.
+        // If the value for "aud" is a String, getAudience() appends the value to the list that is returned.
+        List<String> aud = claims.getAudience();
+        for (String item: aud) {
+            if (item.startsWith(APIConstants.JwtTokenConstants.ENV_NAME_PREFIX)) {
+                String choreoGatewayEnv = APIConstants.JwtTokenConstants.ENV_NAME_PREFIX + gatewayEnv;
+                if (aud.contains(choreoGatewayEnv)) {
+                    log.debug("Environment validation for the access token was successful.");
+                } else {
+                    log.debug("The access token does not have access to the environment {}.", gatewayEnv);
+                    throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
+                            APISecurityConstants.API_AUTH_INVALID_ENVIRONMENT,
+                            APISecurityConstants.API_AUTH_INVALID_ENVIRONMENT_ERROR_MESSAGE);
+                }
+                break;
```

Comment:
Instead of looping through all env values, we first of all create the env value we will be expecting in the array. Then when checking the aud array, as soon as we come across the first value that has the prefix `choreo:environment:`, we directly check if "choreo:environment:<our_env_name>" in there in the array.

Let me add more comments.